server/midddleware/middleware.go

```diff
@@ -26,17 +26,26 @@ import (
 	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/recovery"
 	"github.com/rs/zerolog/log"
 	"github.com/tigrisdata/tigris/server/config"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
 	"google.golang.org/grpc"
 	grpctrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/google.golang.org/grpc"
 )
 
-func Get(config *config.Config) (grpc.UnaryServerInterceptor, grpc.StreamServerInterceptor) {
+func Get(config *config.Config, tenantMgr *metadata.TenantManager, txMgr *transaction.Manager) (grpc.UnaryServerInterceptor, grpc.StreamServerInterceptor) {
 	jwtValidator := GetJWTValidator(config)
 	// inline closure to access the state of jwtValidator
 	authFunction := func(ctx context.Context) (context.Context, error) {
 		return AuthFunction(ctx, jwtValidator, config)
 	}
 
+	namespaceInitializer := NamespaceInitializer{
```

Comment:
is it an `Initializer`? or is it more like a `setter`? should we call it `namespaceSetter`